tests/run-make/const-prop-lint/rmake.rs

```diff
@@ -0,0 +1,19 @@
+// Test that emitting an error because of arithmetic overflow lint does not leave .o files around
```

Comment:
Nit: I think this test is trying to check more like "const prop lints interrupting codegen don't leave .o files around"?

Reply:
Better indeed!